x-pack/plugins/task_manager/server/task_poller.ts

```diff
@@ -30,12 +30,12 @@ import { timeoutPromiseAfter } from './lib/timeout_promise_after';
 type WorkFn<T, H> = (...params: T[]) => Promise<H>;
 
 interface Opts<T, H> {
-  pollInterval: number;
+  pollInterval$: Observable<number>;
   bufferCapacity: number;
   getCapacity: () => number;
   pollRequests$: Observable<Option<T>>;
   work: WorkFn<T, H>;
-  workTimeout?: number;
+  workTimeout: number;
```

Comment:
Turns out only the tests weren't passing this value. It became simpler to make it a required parameter now that `pollInterval` is an observable.

Reply:
Fair enough 👍